doc/basic_types.md

```diff
@@ -1,5 +1,360 @@
+```
+
+Returns `true` if the underlying value is a JavaScript `Number` or `false`
+otherwise.
+
+#### IsString
+```cpp
+bool IsString() const;
+```
+
+Returns `true` if the underlying value is a JavaScript `String` or `false`
+otherwise.
+
+#### IsSymbol
+```cpp
+bool IsSymbol() const;
+```
+
+Returns `true` if the underlying value is a JavaScript `Symbol` or `false`
+otherwise.
+
+#### IsArray
+```cpp
+bool IsArray() const;
+```
+
+Returns `true` if the underlying value is a JavaScript `Array` or `false`
+otherwise.
+
+#### IsArrayBuffer
+```cpp
+bool IsArrayBuffer() const;
+```
+
+Returns `true` if the underlying value is a JavaScript `ArrayBuffer` or `false`
+otherwise.
+
+#### IsTypedArray
+```cpp
+bool IsTypedArray() const;
+```
+
+Returns `true` if the underlying value is a JavaScript `TypedArray` or `false`
+otherwise.
+
+#### IsObject
+```cpp
+bool IsObject() const;
+```
+
+Returns `true` if the underlying value is a JavaScript `Object` or `false`
+otherwise.
+
+#### IsFunction
+```cpp
+bool IsFunction() const;
+```
+
+Returns `true` if the underlying value is a JavaScript `Function` or `false`
+otherwise.
+
+#### IsPromise
+```cpp
+bool IsPromise() const;
+```
+
+Returns `true` if the underlying value is a JavaScript `Promise` or `false`
+otherwise.
+
+#### IsDataView
+```cpp
+bool IsDataView() const;
+```
+
+Returns `true` if the underlying value is a JavaScript `DataView` or `false`
+otherwise.
+
+#### IsBuffer
+```cpp
+bool IsBuffer() const;
+```
+
+Returns `true` if the underlying value is a Node `Buffer` or `false`
+otherwise.
+
+#### IsExternal
+```cpp
+bool IsExternal() const;
+```
+
+Returns `true` if the underlying value is a N-API external object or `false`
+otherwise.
+
+#### ToBoolean
+```cpp
+Boolean ToBoolean() const;
+```
+
+Converts a `Napi::Value` to a JavaScript boolean value. This is a wrapper 
+around `napi_coerce_to_boolean`. This will throw a JavaScript exception if the 
+coercion fails.
+
+#### ToNumber
+```cpp
+Number ToNumber() const;
+```
+
+Converts a `Napi::Value` to a JavaScript Number value. Note that this can cause
+script code to be executed according to JavaScript semantics. This is a wrapper 
+around `napi_coerce_to_number`. This will throw a JavaScript exception if the 
+coercion fails.
+
+
+#### ToString
+```cpp
+String ToString() const;
+```
+
+Converts a `Napi::Value` to a JavaScript String value. Note that this can cause
+script code to be executed according to JavaScript semantics. This is a wrapper 
+around `napi_coerce_to_string`. This will throw a JavaScript exception if the 
+coercion fails.
+
+#### ToObject
+```cpp
+Object ToObject() const;
+```
+
+Converts a `Napi::Value` to a JavasScript Object. This is a wrapper around
+`napi_coerce_to_object`. This will throw a JavaScript exception if the 
+coercion fails.
+
+## Name
+
+Names are JavaScript values that can be used as a property name. There are two
+specialized types of names supported in Node Addon API- `String` and `Symbol`.
```

Comment:
Maybe you can link the doc of String and Symbol:
Names are JavaScript values that can be used as a property name. There are two
specialized types of names supported in Node Addon API- **[String](string.md)** and **[Symbol](symbol.md)**.